docs/source/scancode-toolkit/cli-reference/core-options.rst

```diff
@@ -0,0 +1,282 @@
+
+    ./scancode -clpieu --json-pp sample_generated.json samples --generated
+
+In the results, for each file the following attribute is added with it's corresponding True/False
+value ::
+
+    "is_generated": true
+
+For the samples folder, the following files have a true value for their is_generated attribute::
+
+    "samples/zlib/dotzlib/LICENSE_1_0.txt"
+    "samples/JGroups/licenses/apache-2.0.txt"
+
+..
+    [ToDo] Research and Write Better
+
+``--max-email`` Options
+-----------------------
+
+If in the files that are scanned, in individual files, there are a lot of emails (i.e lists) which
+are unneccesary and clutter the scan results, ``--max-email`` option can be used to report emails
+only up to a limit in individual files.
+
+Some important INTEGER values of the ``--max-email INTEGER`` option:
+
+- 0  - No limit, inlcude all emails.
+- 50 - Default.
+
+An example usage::
+
+    ./scancode -clieu --json-pp sample.json samples --max-email 5
+
+This only reports 5 email adresses per file and ignores the rest.
+
+``--max-url`` Options
+---------------------
+
+If in the files that are scanned, in individual files, there are a lot of links to other websites
+(i.e url lists) which are unneccesary and clutter the scan results, ``--max-url`` option can be
+used to report urls only up to a limit in individual files.
+
+Some important INTEGER values of the ``--max-url INTEGER`` option:
+
+- 0  - No limit, inlcude all urls.
+- 50 - Default.
+
+An example usage::
+
+    ./scancode -clieu --json-pp sample.json samples --max-url 10
+
+This only reports 10 urls per file and ignores the rest.
+
+``--license-score`` Options
+---------------------------
+
+..
+    [ToDo] Research and Write License Matching Better
+
+License matching strictness, i.e. how closely matched licenses are detected in a scan, can be
+modified by using this ``--license-score`` option.
+
+Some important INTEGER values of the ``--license-score INTEGER`` option:
+
+- **0**     - Default and Lowest Value, All matches are reported.
+- **100**    - Highest Value, Only licences with a much better match are reported
+
+Here, a bigger number means a better match, i.e. setting a higher license score translates to a
+higher threshold for matching licenses (with equal or less number of license matches).
+
+An example usage::
+
+    ./scancode -clieu --json-pp sample.json samples --license-score 70
+
+Here's the license results on setting the integer value to 100, vs the default value 0. This is
+visualized using ScanCode workbench in the License Info Dashboard.
+
+.. list-table:: License scan results of Samples Directory.
+
+    * - .. figure:: data/core_lic_score_0.png
+
+           License Score 0 (Default).
+
+      - .. figure:: data/core_lic_score_100.png
+
+           License Score 100.
+
+``--license-text`` Options
+--------------------------
+
+The ``--license-text`` option includes in the scan result, the matched text for the detected
```

Comment:
This is a general comment on the way to craft some sentences:
Here 

> The ``--license-text`` option includes in the scan result, the matched text for the detected
license, under the attribute "matched text".

may be easier to read as:

> With the ``--license-text`` option, the scan results attribute "matched text" includes the matched text for the detected
license.


@DennisClark what do you think? I am not a native speaker of English!

Reply:
This is really an important issue. I'll look into this and correct stuff all across the docs. Suggestions and help on this is much appreciated.

Reply:
Or something like "`--license-text`: Include the matched licensed text in the scan results for each detected license."

Reply:
After a join chat today, no major change needed, though it is better to use direct vs. indirect style as general guideline.